esr/esr_release.py

```diff
@@ -0,0 +1,493 @@
+        logger.info ("Deploying %s ESR version %s to %s" % (device, esr_tag, self.arg_dict['deploy-environment']))
+        requests.post(url, auth=(self.arg_dict['jenkins-user'], self.arg_dict['jenkins-token']), data=params)
+
+    def build_device_types_json(self):
+        process = Popen("./balena-yocto-scripts/build/build-device-type-json.sh", stdout=PIPE, stderr=PIPE)
+        stdout, stderr = process.communicate()
+
+    def set_device_types(self, version):
+        files = []
+        self.build_device_types_json()
+        files += [each for each in os.listdir(os.getcwd()) if each.endswith(".json")]
+        for f in files:
+            with open(f, "r") as fd:
+                jobj = json.load(fd)
+                self.device_list.update({jobj['slug']: version})
+
+    def fetch_canonical_device_list(self, token=None):
+        apiEnv = "balena-cloud.com/"
+        translation = "v6"
+        url = "https://api." + apiEnv + translation + "/device_type?$select=slug"
+        header = {}
+        if token != None:
+            header = {'Authorization': "bearer " + token}
+        raw = urllib.request.urlopen(urllib.request.Request(url, headers=header))
+        json_obj = json.load(raw)
+        for e in json_obj['d']:
+            self.canonical_device_list.append(e['slug'])
+
+def usage():
+    print ("Usage: " + script_name + " [OPTIONS]\n")
+    print ("\t-h Display usage")
+    print ("\t-e ESR version (e.g 2021.01). Required.")
+    print ("\t-b BalenaOS version (e.g 2.68). Required.")
+    print ("\t-u Jenkins user name. If provided along with a Jenkins API token, deploy jobs for the specified environment will be run.")
+    print ("\t-t Jenkins API token. If provided along with a Jenkins username, deploy jobs for the specified environment will be run.")
+    print ("\t-d Balena deploy environment (defaults to staging).")
+    print ("\t-a Balena API token with access to balena_os organization. If provided, private device types will be considered for deploy jobs.")
+    print ("\t-g Github API token (for meta-balena https git access). If omitted, username and password credentials will be prompted for.")
+    print ("\t-k Keep temporary directory (testing only).\n")
+    sys.exit(1)
+
+def main(argv):
+    logger.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    lf = logging.FileHandler(os.path.basename(__file__) + str(int(time.time())) + '.log')
+    lf.setLevel(logging.DEBUG)
+    lf.setFormatter(formatter)
+    logger.addHandler(lf)
+    ch = logging.StreamHandler()
+    ch.setLevel(logging.INFO)
+    ch.setFormatter(formatter)
+    logger.addHandler(ch)
+
+    try:
+        opts, args = getopt(argv[1:], "he:b:u:t:d:a:g:k", ["help", "esr-version=", "bos-version=", "user", "token", "deploy-to", "api-token", "github-api-token", "keep-tmp-dir="])
```

Comment:
Shouldn't the long style options that require an argument be suffixed with `=`?

Reply:
just the ones that require an argument, but I missed some :) thanks